Lib/zipfile.py

```diff
@@ -1632,6 +1633,29 @@ def extractall(self, path=None, members=None, pwd=None):
 
         for zipinfo in members:
             self._extract_member(zipinfo, path, pwd)
+            
+    def remove(self, member):
+        """Remove a file from the archive. The archive must be open with mode 'a'"""
+
+        if self.mode != 'a':
+            raise RuntimeError("remove() requires mode 'a'")
+        if not self.fp:
+            raise ValueError(
+                "Attempt to write to ZIP archive that was already closed")
+        if self._writing:
+            raise ValueError(
+                "Can't write to ZIP archive while an open writing handle exists."
+            )
+
+        # Make sure we have an info object
+        if isinstance(member, ZipInfo):
+            # 'member' is already an info object
+            zinfo = member
```

Comment:
This is unsafe.  The `ZipInfo` instances in `self.filelist` will not compare equal with this zinfo even if the filename is matching. They compare by identity.